src/data/W2lListFilesDataset.cpp

```diff
@@ -78,7 +78,12 @@ std::vector<W2lLoaderData> W2lListFilesDataset::getLoaderData(
     }
 
     data[id].sampleId = data_[i].getSampleId();
-    data[id].input = loadSound(data_[i].getAudioFile());
+    auto path = data_[i].getAudioFile();
+    try {
+        data[id].input = loadSound(path);
+    } catch (const std::exception &ex) {
+        throw std::runtime_error("loadSound failed: " + path + " | " + ex.what());
```

Comment:
Please add context to the exception to make for faster debugging.
Also the path should already be included in the exception returned from loadSound.
see https://github.com/facebookresearch/wav2letter/blob/4289710cd65f03722d6d0d20d801e82290823408/src/data/Sound.cpp#L226

Suggesting.
throw std::runtime_error("lW2lListFilesDataset::getLoaderData(idx=" + std::to_string(idx) + ") failed to load sound file with error=" + ex.what());